commit 51ead00bf81cd92a868bdd1551a06ec5efdb563b
Author: Fangrui Song <maskray@google.com>
Date:   Fri Sep 13 02:18:04 2019 +0000

    [ELF] Delete a redundant assignment to SectionBase::assigned. NFC
    
    LinkerScript::discard marks a section dead. It is unnecessary to set the
    `assigned` bit.
    
    llvm-svn: 371804

diff --git a/lld/ELF/LinkerScript.cpp b/lld/ELF/LinkerScript.cpp
index dbf705dc7f0..8d8f3b596fa 100644
--- a/lld/ELF/LinkerScript.cpp
+++ b/lld/ELF/LinkerScript.cpp
@@ -457,7 +457,6 @@ void LinkerScript::discard(ArrayRef<InputSection *> v) {
     if (s == mainPart->hashTab)
       mainPart->hashTab = nullptr;
 
-    s->assigned = false;
     s->markDead();
     discard(s->dependentSections);
   }
